<?xml version="1.0"?>
<robot name="robot">

  <material name="red">
    <color rgba="0.8 0 0 1"/>
  </material>

  <material name="white">
    <color rgba="1 1 1 0.5"/>
  </material>

<link name="base_link">
</link>

  <link name="link_0">
    <visual>
      <geometry>
        <box  size ="0.3 0.4 0.2 "/>
      </geometry>
      <material name="white"/>
      <origin rpy="0 0 0" xyz="0 0 0.1"/>
    </visual>
  </link>

<joint name="base_joint" type="fixed">
  <parent link="base_link"/>
  <child link="link_0"/>
  <origin xyz="0 0 0"/>
</joint>

  <link name="link_1">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.12"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <material name="red"/>
    </visual>
  </link>

  <joint name="jiont_0" type="revolute">
    <parent link="link_0"/>
    <child link="link_1"/>
    <origin xyz="0 0 0.2"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" velocity="1.0" lower="-0.5" upper="1"/>
  </joint>

  <link name="link_2">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.4"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <material name="white"/>
    </visual>
  </link>

  <joint name="joint_1" type="revolute">
    <parent link="link_1"/>
    <child link="link_2"/>
    <origin xyz="0 0 0.1"/>
    <axis xyz="1 0 0"/>
    <limit effort="100" velocity="1.0" lower="-0.5" upper="1"/>
  </joint>

  <link name="link_3">
    <visual>
      <geometry>
        <box size="0.08 0.08 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <material name="red"/>
    </visual>
  </link>

  <joint name="joint_2" type="revolute">
    <parent link="link_2"/>
    <child link="link_3"/>
    <origin xyz="0 0 0.4"/>
    <axis xyz="1 0 0"/>
    <limit effort="100" velocity="1.0" lower="-0.5" upper="1"/>
  </joint>


</robot>
